packages/ui/src/common/providers/modal/provider.tsx

```diff
@@ -26,7 +27,14 @@ export const ModalContextProvider = (props: Props) => {
     },
     hideModal: (showConfirmOrAny = false) => {
       const preferShowConfirm = typeof showConfirmOrAny === 'boolean' ? showConfirmOrAny : false
-      if (!preferShowConfirm || isClosing || !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames)) {
+      const [state] = currentModalMachine ?? []
+      if (
+        !preferShowConfirm ||
+        isClosing ||
+        !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames) ||
+        state?.matches('success') ||
+        state?.matches('error')
```

Comment:
We want to keep the modal on errors to retry or save inputs. Only hide on success not if [transaction was not submitted](https://github.com/Joystream/pioneer/issues/3899#issuecomment-1335247993).